elasticai/creator/vhdl/generator/precomputed_scalar_function.py

```diff
@@ -82,17 +77,24 @@ def __call__(self) -> list[str]:
             "x : in signed(DATA_WIDTH-1 downto 0)",
             "y : out signed(DATA_WIDTH-1 downto 0)",
         ]
-        code = "\n".join(chain([get_libraries_string()], chain(entity()), [""]))
-        code += get_architecture_header_string(
-            architecture_name=self.architecture_name, component_name=self.component_name
-        )
-        code += get_process_string(
-            component_name=self.component_name,
+        library = Library()
+        code = "\n".join(chain(chain(library()), chain(entity()), [""]))
+        process = Process(
+            identifier=self.component_name,
             lookup_table_generator_function=precomputed_scalar_function_process(
                 x_list=self.x, y_list=self.y
             ),
+            input="x",
+        )
+        process.variable_initialization_list = ["variable int_x: integer := 0"]
+        process.variable_list = ["int_x := to_integer(x)"]
+        process_code = "\n".join(chain(chain(process()), [""]))
+        architecture = Architecture(
+            identifier=self.architecture_name,
+            design_unit=self.component_name,
+            process_content=process_code,
         )
-        code += get_architecture_end_string(architecture_name=self.architecture_name)
+        code += "\n".join(chain(chain(architecture()), [""]))
         return [code]
```

Comment:
I think it's best to skip all the string concatenation above and instead have this class return an iterable of strings, where each string corresponds to one line of source code again. It will then be the task of whatever creates e.g. a file to write the lines correctly.

Reply:
I resolved it but I don't know if it is implemented how you had it in mind